snapshot/lib/snapshot/test_command_generator.rb

```diff
@@ -40,8 +40,12 @@ def destination(devices)
 
         destinations = devices.map do |d|
           device = find_device(d, os_version)
-          UI.user_error!("No device found named '#{d}' for version '#{os_version}'") if device.nil?
-          "-destination 'platform=#{os} Simulator,name=#{device.name},OS=#{os_version}'"
+          if device.nil?
+            UI.user_error!("No device found named '#{d}' for version '#{os_version}'") if device.nil?
+          elsif device.os_version != os_version
```

Comment:
How does this work? If the device is found from `find_device`, shouldnt the os_versions match?

Reply:
find_device gives preference to the os_version that matches, if not it takes the highest (per Gem::Version definition) version with the same device name